collection+xattrs.go

```diff
@@ -39,6 +39,69 @@ func (c *Collection) GetXattr(
 	}
 }
 
+// SetWithMeta updates a document fully with xattrs and body and allows specification of a specific CAS (newCas). This update will always happen as long as oldCas matches the value of existing document. This simulates the kv op setWithMeta.
+func (c *Collection) SetWithMeta(_ context.Context, key string, oldCas CAS, newCas CAS, exp uint32, xattrs []byte, body []byte, isJSON bool) error {
```

Comment:
Do you think it's reasonable that callers are expected to know what the isJSON flag represents and set it?  This is basically equivalent to dataType for server's setWithMeta - should we be accepting something like FeedDataType here instead, or do you think it's appropriate for callers to convert the FeedDataType they get on the FeedEvent to this flag?